### PR TITLE
[NVIDIA] Relax the requirement for providing both `query_seq_lengths` and `key_value_seq_lengths`

### PR DESCRIPTION
**Pull request by kaixih**
This PR addresses an issue where the original `nn.dot_product_attention` required both `query_seq_lengths` and `key_value_seq_lengths`. With this update, users can now provide only one of these lengths, thereby relaxing the previous requirement.

This is motivated by [this issue](https://github.com/google/jax/issues/23349).

cc. @superbobry

### CODE REVIEW

**Comment by kaixih** [comment]
Also, @sbodenstein for review.

File: jax/_src/nn/functions.py [hunk 1]
```diff
@@ -786,10 +786,14 @@ def _get_causal_mask(T, S):
   return mask[None, None, :, :]
 
 def _get_padding_mask_logits(T, S, q_seqlen, kv_seqlen):
-  q_indices = jnp.arange(0, T)[None, :, None]
-  kv_indices = jnp.arange(0, S)[None, None, :]
-  q_mask = q_indices < q_seqlen[:, None, None]
-  kv_mask = kv_indices < kv_seqlen[:, None, None]
+  q_mask = jnp.array(True, dtype=jnp.bool_)
```
Comment (superbobry) [inline jax/_src/nn/functions.py:789]: Can you do `jnp.bool_(True)` or just `True` here?

File: tests/nn_test.py [hunk 2]
```diff
@@ -122,7 +122,6 @@ def testDotProductAttentionMask(self, mask_mode):
 
     is_causal = 'causal' in mask_mode
     if 'padding' in mask_mode:
-      q_seqlen = jnp.array([T // 2, T // 4], dtype=jnp.int32)
```
Comment (superbobry) [inline tests/nn_test.py:125]: Is this change necessary?

**Review by superbobry — CHANGES_REQUESTED** [review]
Can you add a regression test for #23349, please?

File: tests/nn_test.py (hunk 2 above)
Comment (kaixih) [reply tests/nn_test.py]: Done.

File: jax/_src/nn/functions.py (hunk 1 above)
Comment (kaixih) [reply jax/_src/nn/functions.py]: Changed.

**Comment by sbodenstein** [comment]
LGTM

**Review by superbobry — APPROVED** [review]
